lib/rules/no-param-reassign.js

```diff
@@ -22,7 +22,13 @@ module.exports = {
             {
                 type: "object",
                 properties: {
-                    props: { type: "boolean" }
+                    props: { type: "boolean" },
+                    ignorePropertyAssignmentsFor: {
```

Comment:
Thanks for the feedback! I just pushed 481692a to restrict the schema, please let me know if you have any other suggestions. I didn't realize that `oneOf` or `enum` were even options, but they seem like a perfect fit for this use-case.